warehouse/query-core/src/test/java/datawave/query/tables/RemoteEventQueryLogicIT.java

```
@@ -0,0 +1,273 @@
+import datawave.webservice.common.remote.RemoteHttpService;
+import datawave.webservice.common.remote.RemoteHttpServiceConfiguration;
+import datawave.webservice.query.remote.RemoteQueryServiceImpl;
+
+@Category(IntegrationTest.class)
+public class RemoteEventQueryLogicIT {
+    private RemoteEventQueryLogic logic = new RemoteEventQueryLogic();
+    private RemoteQueryServiceTestUtil testUtil;
+
+    private UUID uuid;
+    private AtomicBoolean handlerInterrupt = new AtomicBoolean(false);
+
+    @Before
+    public void setup() throws Exception {
+        uuid = UUID.randomUUID();
+        handlerInterrupt.set(false);
+        testUtil = new RemoteQueryServiceTestUtil(uuid, DEFAULT_REMOTE_LOGIC, 3);
+        testUtil.initialize();
+
+        // create a remote event query logic that has our own server behind it
+        RemoteQueryServiceImpl remote = (RemoteQueryServiceImpl) testUtil.getRemoteService();
+
+        logic.setRemoteQueryService(remote);
+        logic.setRemoteQueryLogic(DEFAULT_REMOTE_LOGIC);
+    }
+
+    @After
+    public void after() {
+        handlerInterrupt.set(true);
+        testUtil.stop();
+    }
+
+    @Test
+    public void testRemoteQuery() {
+        RemoteQueryServiceTestUtil.QueryRunnable r = new RemoteQueryServiceTestUtil.QueryRunnable(logic, testUtil);
+        r.run();
+    }
+
+    @Test
+    public void testDefaultConnectTimeoutHangForever() {
+        // override the endpoint to a non-routable ip, so it will block forever
+        ((RemoteHttpService) logic.getRemoteQueryService()).setQueryServiceHost(NON_ROUTABLE_HOST);
+
+        RemoteQueryServiceTestUtil.QueryRunnable r = new RemoteQueryServiceTestUtil.QueryRunnable(logic);
+        Thread t = new Thread(r);
+        t.start();
+
+        // ensure the thread wasn't interrupted
+        boolean interrupted = false;
+        try {
+            // waiting forever is the default state
+            t.join(1000);
```

periodic test failure potential

---

This should probably be quite a bit larger than 1 second.

---

This test proves the default behavior is to wait indefinitely. Anything longer than 10ms is more than enough. I'm going to update this to block for 100ms, but also put a while loop around it that silently catches an interrupted exception and waits so we have a guarantee of 100ms of waiting. The validation of the thread state is really what is important here.